### ISSUE TITLE
Make failed on main branch

### ISSUE THREAD

**Issue by firesunny0** [lwfinger/rtw89#194]
This driver helped me **solve the problem (Ubuntu 18.04, 8852AE)**. But when I first used this driver, I found that it could not be compiled. Retrieve the experience of others and find that the v5 branch can solve this problem, but it was merged into the main branch a few days ago. Finally, it runs successfully by r**olling back the v5 branch to the commit before merged to main,** finishing compiling and installing.
      The main branch  report the following error when "make":
```
In file included from /home/xxxxx/tmp/rtw89/cam.h:8:0,
                 from /home/xxxxx/tmp/rtw89/core.c:7:
/home/xxxxx/tmp/rtw89/core.h:4431:18: warning: ‘struct cfg80211_tid_config’ declared inside parameter list will not be visible outside of this definition or declaration
           struct cfg80211_tid_config *tid_config);
                  ^~~~~~~~~~~~~~~~~~~
/home/xxxxx/tmp/rtw89/core.c:2609:19: warning: ‘struct cfg80211_tid_cfg’ declared inside parameter list will not be visible outside of this definition or declaration
            struct cfg80211_tid_cfg *tid_conf)
                   ^~~~~~~~~~~~~~~~
/home/xxxxx/tmp/rtw89/core.c: In function ‘_rtw89_core_set_tid_config’:
/home/xxxxx/tmp/rtw89/core.c:2613:21: error: dereferencing pointer to incomplete type ‘struct cfg80211_tid_cfg’
  u32 mask = tid_conf->mask;
                     ^~
In file included from ./include/linux/bitops.h:5:0,
                 from ./include/linux/kernel.h:12,
                 from ./include/linux/skbuff.h:13,
                 from ./include/linux/ip.h:16,
                 from /home/xxxxx/tmp/rtw89/core.c:4:
/home/xxxxx/tmp/rtw89/core.c:2628:18: error: ‘NL80211_TID_CONFIG_ATTR_AMPDU_CTRL’ undeclared (first use in this function); did you mean ‘NL80211_MESHCONF_ATTR_MAX’?
   if (mask & BIT(NL80211_TID_CONFIG_ATTR_AMPDU_CTRL)) {
                  ^
./include/linux/bits.h:8:30: note: in definition of macro ‘BIT’
 #define BIT(nr)   (UL(1) << (nr))
                              ^~
/home/xxxxx/tmp/rtw89/core.c:2628:18: note: each undeclared identifier is reported only once for each function it appears in
   if (mask & BIT(NL80211_TID_CONFIG_ATTR_AMPDU_CTRL)) {
                  ^
./include/linux/bits.h:8:30: note: in definition of macro ‘BIT’
 #define BIT(nr)   (UL(1) << (nr))
                              ^~
/home/xxxxx/tmp/rtw89/core.c:2629:27: error: ‘NL80211_TID_CONFIG_ENABLE’ undeclared (first use in this function); did you mean ‘NL80211_CMD_CONN_FAILED’?
    if (tid_conf->ampdu == NL80211_TID_CONFIG_ENABLE) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~
                           NL80211_CMD_CONN_FAILED
/home/xxxxx/tmp/rtw89/core.c: At top level:
/home/xxxxx/tmp/rtw89/core.c:2654:18: warning: ‘struct cfg80211_tid_config’ declared inside parameter list will not be visible outside of this definition or declaration
           struct cfg80211_tid_config *tid_config)
                  ^~~~~~~~~~~~~~~~~~~
/home/xxxxx/tmp/rtw89/core.c:2652:6: error: conflicting types for ‘rtw89_core_set_tid_config’
 void rtw89_core_set_tid_config(struct rtw89_dev *rtwdev,
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/xxxxx/tmp/rtw89/cam.h:8:0,
                 from /home/xxxxx/tmp/rtw89/core.c:7:
/home/xxxxx/tmp/rtw89/core.h:4429:6: note: previous declaration of ‘rtw89_core_set_tid_config’ was here
 void rtw89_core_set_tid_config(struct rtw89_dev *rtwdev,
      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/xxxxx/tmp/rtw89/core.c: In function ‘rtw89_core_set_tid_config’:
/home/xxxxx/tmp/rtw89/core.c:2658:28: error: dereferencing pointer to incomplete type ‘struct cfg80211_tid_config’
  for (i = 0; i < tid_config->n_tid_conf; i++)
                            ^~
/home/xxxxx/tmp/rtw89/core.c: In function ‘rtw89_core_register_hw’:
/home/xxxxx/tmp/rtw89/core.c:3357:13: error: ‘struct wiphy’ has no member named ‘tid_config_support’; did you mean ‘bss_select_support’?
  hw->wiphy->tid_config_support.vif |= BIT(NL80211_TID_CONFIG_ATTR_AMPDU_CTRL);
             ^~~~~~~~~~~~~~~~~~
             bss_select_support
In file included from ./include/linux/bitops.h:5:0,
                 from ./include/linux/kernel.h:12,
                 from ./include/linux/skbuff.h:13,
                 from ./include/linux/ip.h:16,
                 from /home/xxxxx/tmp/rtw89/core.c:4:
/home/xxxxx/tmp/rtw89/core.c:3357:43: error: ‘NL80211_TID_CONFIG_ATTR_AMPDU_CTRL’ undeclared (first use in this function); did you mean ‘NL80211_MESHCONF_ATTR_MAX’?
  hw->wiphy->tid_config_support.vif |= BIT(NL80211_TID_CONFIG_ATTR_AMPDU_CTRL);
                                           ^
./include/linux/bits.h:8:30: note: in definition of macro ‘BIT’
 #define BIT(nr)   (UL(1) << (nr))
                              ^~
/home/xxxxx/tmp/rtw89/core.c:3358:13: error: ‘struct wiphy’ has no member named ‘tid_config_support’; did you mean ‘bss_select_support’?
  hw->wiphy->tid_config_support.peer |= BIT(NL80211_TID_CONFIG_ATTR_AMPDU_CTRL);
             ^~~~~~~~~~~~~~~~~~
             bss_select_support
make[2]: *** [/home/xxxxx/tmp/rtw89/core.o] Error 1
make[1]: *** [/home/xxxxx/tmp/rtw89] Error 2
make: *** [all] Error 2
```

**Comment by lwfinger** [comment]
The driver in branch v5 is definitely inferior to the one in branch main; however, the latter will never be transformed to build with kernels older than v5.0. I would suggest updating your kernel, but if you are happy with the v5 branch, keep using it.
